src/v/cloud_topics/dl_placeholder.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_topics/types.h"
+#include "serde/envelope.h"
+#include "serde/rw/named_type.h"
+#include "serde/rw/uuid.h"
+
+// This header contains definition of the dl_placeholder batch
+
+namespace cloud_topics {
+
+struct dl_placeholder // NOLINT
```

Comment:
what does the `dl_` prefix mean?

Reply:
this means `data-layout`
it's not consumed by data layout stm at this point but it could be and it's the same area so I decided to use common prefix